src/pydis_jam23/codecs/__init__.py

```diff
@@ -20,3 +19,30 @@ def decode(self, image: Image.Image) -> bytes:
 
 
 CODECS: list[Codec] = [lsb]
+
+
+def encode_message(plain: BinaryIO, codec: Codec, gui_data: str | None = None) -> None:
+    if gui_data is None:
+        message = sys.stdin.buffer.read()
+        image = Image.open(plain)
+        codec.encode(image, message)
+        image.save(sys.stdout.buffer, format="PNG")
+    else:
+        message = gui_data.encode("utf-8")  # ui_app message
+        image = Image.open(plain)
+        codec.encode(image, message)
+        return image
+
+
+def decode_message(extract: BinaryIO, codec: Codec, gui_data: str | None = None) -> None:
+    if gui_data is None:
+        image = Image.open(extract)
+        message = codec.decode(image)
+        sys.stdout.buffer.write(message)
+    else:
+        image = Image.open(gui_data)  # image file
+        message = codec.decode(image)
+        return message
```

Comment:
These appear to be two separate functions each, why combine them? It is probably better for code which is specific to the CLI or GUI to live with the code for those systems rather than here.